physionet-django/events/views.py

```diff
@@ -98,43 +98,50 @@ def event_home(request):
-                    if form.is_valid():
-                        event_application = form.save(commit=False)
-                        if event_application.status == EventApplication.EventApplicationStatus.APPROVED:
-                            event_application.accept(
-                                comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
-                            )
-                            notification.notify_participant_event_decision(
-                                request=request,
-                                user=event_application.user,
-                                event=event_application.event,
-                                decision=EventApplication.EventApplicationStatus.APPROVED.label,
-                                comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
-                            )
-                        elif event_application.status == EventApplication.EventApplicationStatus.NOT_APPROVED:
-                            event_application.reject(
-                                comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
-                            )
-                            notification.notify_participant_event_decision(
-                                request=request,
-                                user=event_application.user,
-                                event=event_application.event,
-                                decision=EventApplication.EventApplicationStatus.NOT_APPROVED.label,
-                                comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
-                            )
-                    else:
-                        messages.error(request, form.errors)
-                    return redirect(event_home)
-            else:
-                form_error = True
+    # handle notifications to join an event
+    if request.method == 'POST' and 'participation_response' in request.POST.keys():
+
+        formset = EventApplicationResponseFormSet(request.POST)
+        # only process the form that was submitted
+        for form in formset:
+            if form.instance.id != int(request.POST['participation_response']):
+                continue
+
+            if not form.is_valid():
+                messages.error(request, form.errors)
+                return redirect(event_home)
+
+            event_application = form.save(commit=False)
+            event = event_application.event
+            if event.host != user:
+                messages.error(request, "You don't have permission to accept/reject this application")
+                return redirect(event_home)
+            elif event_application.status == EventApplication.EventApplicationStatus.APPROVED:
+                event_application.accept(
+                    comment_to_applicant=form.cleaned_data.get('comment_to_applicant')
+                )
+                notification.notify_participant_event_decision(
```

Comment:
I feel like there is probably a way to structure this to avoid duplicated `notify_participant_event_decision` code but this is fine for now

Reply:
I agree, let me make a note and open a separate PR for that.